src/cmake/CommonSettings.cmake

```diff
@@ -3,6 +3,35 @@ function(set_common_settings)
     set(CMAKE_CXX_STANDARD_REQUIRED ON PARENT_SCOPE)
     set(CMAKE_EXPORT_COMPILE_COMMANDS ON PARENT_SCOPE)
 
+    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

Comment:
This will be called for every target that uses this function, but it doesn't seem to be needed... maybe we should create a separate file with a new function to do the installation related setup, and have this function only be called from the root cmake file.